src/Shouldly/ShouldlyExtensionMethods/ShouldBe/GenericShouldBeTestExtensions.cs

```diff
@@ -57,11 +58,8 @@ public static void ShouldNotBe<T>(this T? actual, T? expected, IEqualityComparer
             bool ignoreOrder,
             string? customMessage)
         {
-            if (actual is not (null or IReadOnlyCollection<T> or ICollection<T> or ICollection))
-                actual = actual.ToList();
-
-            if (expected is not (null or IReadOnlyCollection<T> or ICollection<T> or ICollection))
-                expected = expected.ToList();
+            actual = actual.Wrap();
+            expected = expected.Wrap();
```

Comment:
I could get rid of the extension and use `EnumerableProxy<T>.Wrap` here instead, unless you prefer to repeat the checks wherever the Proxy is used.

Reply:
I'm sorry, I misunderstood the first few times I came to this code and thought that the `Wrap` method always wrapped. If you call it `WrapIfNonCollection` then I don't mind the checks moving into that method.

Reply:
Actually, collections are wrapped as well, I just don't call `ToList` on them. The only time no wrapping occurs is if the value is null or if it is already wrapped. The idea was that you could call `Wrap` at all entry points, without having to worry about excessive wrapping if calls are chained.

Reply:
My personal preference if anything would be to throw an exception if Wrap was called on an already-wrapped collection, because if you find yourself wrapping a wrapper, the most likely scenario is that you're making a larger programming mistake.

I'm also not sure I like the concept of wrapping collections (as opposed to only wrapping non-collections), but I feel less strongly there.